src/lib/components/Search/index.js

```diff
@@ -7,18 +7,28 @@ import {BaseElement} from '../BaseElement';
 import {store} from '../../store';
 import * as router from '../../utils/router';
 import {debounce} from '../../utils/debounce';
-import algoliasearch from 'algoliasearch/dist/algoliasearch-lite.esm.browser';
 import {trackError} from '../../analytics';
 import 'focus-visible';
 import './_styles.scss';
 
-// Create an algolia client so we can get search results.
-// These keys are safe to be public.
-const applicationID = '2JPAZHQ6K7';
-const apiKey = '01ca870a3f1cad9984ed72419a12577c';
-const indexName = 'webdev';
-const client = algoliasearch(applicationID, apiKey);
-const index = client.initIndex(indexName);
+let algoliaSearchIndex;
+
+async function loadAlgoliaLibrary() {
+  if (algoliaSearchIndex) {
+    return;
+  }
+
+  const p = await import('algoliasearch/dist/algoliasearch-lite.esm.browser');
+  return p.then(() => {
+    // Create an algolia client so we can get search results.
+    // These keys are safe to be public.
+    const applicationID = '2JPAZHQ6K7';
+    const apiKey = '01ca870a3f1cad9984ed72419a12577c';
+    const indexName = 'webdev';
+    const client = window.algoliasearch(applicationID, apiKey);
```

Comment:
can't we use `algoliasearch` from the resolved value of the `Promise`?

Reply:
Much better indeed.
Fixed in https://github.com/GoogleChrome/web.dev/pull/4175/commits/3e31747f816f83d1dfacbf47a3966bd19528ec48

Reply:
It's not true actually. See https://github.com/GoogleChrome/web.dev/pull/4175#issuecomment-721556584